src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/MemoryMarshal.cs

```diff
@@ -307,7 +307,7 @@ public static bool TryGetArray<T>(ReadOnlyMemory<T> memory, out ArraySegment<T>
         {
             TManager? localManager; // Use register for null comparison rather than byref
             manager = localManager = memory.GetObjectStartLength(out _, out _) as TManager;
-            return localManager != null;
+            return localManager != null; // TODO2
```

Comment:
What's wrong here?

Reply:
The compiler doesn't know that `manager == localManager`, so it looks like we may be returning with a `null` manager when this method returns `true`.
We could fix that by doing `return manager != null;`, but that defeats the purpose indicated by above comment ("Use register for null comparison rather than byref").

Reply:
Ah, so this TODO was about improving the compiler :)

For the purposes of this change, just suppress the warning with whatever syntax is least obtrusive.